account_payment_partner/models/account_move.py

```diff
@@ -0,0 +1,156 @@
+                    ):
+                        move.invoice_partner_bank_id = get_bank_id()
+                    else:
+                        move.invoice_partner_bank_id = False
+                elif (
+                    pay_mode
+                    and pay_mode.payment_type == "outbound"
+                    and not pay_mode.payment_method_id.bank_account_required
+                ):
+                    self.invoice_partner_bank_id = False
+                elif not self.payment_mode_id:
+                    self.invoice_partner_bank_id = False
+            else:
+                if move.type == "in_invoice":
+                    move.invoice_partner_bank_id = False
+
+    @api.depends("type")
+    def _compute_payment_mode_filter_type_domain(self):
+        for move in self:
+            if move.type in ("out_invoice", "in_refund"):
+                move.payment_mode_filter_type_domain = "inbound"
+            elif move.type in ("in_invoice", "out_refund"):
+                move.payment_mode_filter_type_domain = "outbound"
+            else:
+                move.payment_mode_filter_type_domain = False
+
+    @api.depends("partner_id", "type")
+    def _compute_partner_bank_filter_type_domain(self):
+        for move in self:
+            if move.type in ("out_invoice", "in_refund"):
+                move.partner_bank_filter_type_domain = move.bank_partner_id
+            elif move.type in ("in_invoice", "out_refund"):
+                move.partner_bank_filter_type_domain = move.commercial_partner_id
+            else:
+                move.partner_bank_filter_type_domain = False
+
+    payment_mode_filter_type_domain = fields.Char(
+        compute="_compute_payment_mode_filter_type_domain"
+    )
+    partner_bank_filter_type_domain = fields.Many2one(
+        comodel_name="res.partner", compute="_compute_partner_bank_filter_type_domain"
+    )
+    payment_mode_id = fields.Many2one(
+        comodel_name="account.payment.mode",
+        string="Payment Mode",
+        compute=_compute_payment_mode,
+        store=True,
+        readonly=True,
+        ondelete="restrict",
+        states={"draft": [("readonly", False)]},
+    )
+    bank_account_required = fields.Boolean(
+        related="payment_mode_id.payment_method_id.bank_account_required", readonly=True
+    )
+    invoice_partner_bank_id = fields.Many2one(
+        compute=_compute_invoice_partner_bank,
+        store=True,
+        readonly=True,
+        ondelete="restrict",
+        states={"draft": [("readonly", False)]},
+    )
+
+    # I think copying payment mode from invoice to refund by default
+    # is a good idea because the most common way of "paying" a refund is to
+    # deduct it on the payment of the next invoice (and OCA/bank-payment
+    # allows to have negative payment lines since March 2016)
+    @api.model
+    def _reverse_move_vals(self, default_values, cancel=True):
+        move_vals = super(AccountMove, self)._reverse_move_vals(
```

Comment:
You can remove the parameters of the `super()` here and everywhere else in the code.